packages/aws-cdk/lib/context-providers/vpcs.ts

```diff
@@ -168,33 +159,37 @@ function getTag(name: string, tags?: AWS.EC2.Tag[]): string | undefined {
 /**
  * Group subnets of the same type together, and order by AZ
  */
-function groupSubnets(subnets: Subnet[]): SubnetGroups {
-  const grouping: {[key: string]: Subnet[]} = {};
+function groupSubnets(subnets: Subnet[]): cxapi.VpcSubnetGroup[] {
+  const grouping: { [key: string]: Subnet[] } = {};
   for (const subnet of subnets) {
     const key = [subnet.type, subnet.name].toString();
```

Comment:
This confused me to no end, but I suppose this is meant to say `subnet.groupName`, no?

Also, feels like `groupName` should be enough, since the relationship goes `[ type ]<>--*[ name ]<>--*[ subnet ]`.

That is, *same name* already implies *same type* and so it's wasteful to include in the key.

Reply:
> This confused me to no end, but I suppose this is meant to say `subnet.groupName`, no?

It's interesting that you say that, as it was taken verbatim from the [existing VPC provider](https://github.com/aws/aws-cdk/blob/09d89c385375581ba852fb473b191933d09abd0a/packages/aws-cdk/lib/context-providers/vpcs.ts#L174), that I'm pretty sure [you're the author of](https://github.com/aws/aws-cdk/pull/1168/files?file-filters%5B%5D=.ts#diff-d1c15c14f1d5ddd8a752af7579c33871R102) 🙂.

> Also, feels like `groupName` should be enough, since the relationship goes `[ type ]<>--*[ name ]<>--*[ subnet ]`.
> That is, _same name_ already implies _same type_ and so it's wasteful to include in the key.

That's not guaranteed though, right? You can have 2 subnets, one `Private`, one `Public`, with the same group name, right? (If you specify a tag that determines the name)

Reply:
Oh I totally know I'm the author and I wrote bad code while writing this. Please clean up my mess? :rofl: 

> You can have 2 subnets, one `Private`, one `Public`, with the same group name, right?

We would never generate that. In fact, group names have been introduced to disambiguate between different "kinds" of private subnets ("PrivateDatabase", "PrivateAppServer", "PrivateLambda").

I guess the question then becomes, would other people have built this in practice, and should we respect that or reject it?

I don't have a good sense for how this will behave in the rest of the CDK...

Leave it like this then. At the worst it doesn't need to be there, at the best it enables new use cases.